src/core/Badge/Badge.tsx

```diff
@@ -28,13 +29,16 @@ export type BadgeProps = {
  */
 export const Badge = (props: BadgeProps) => {
   const { backgroundColor, style, className, children, ...rest } = props;
+
   useTheme();
+
+  const _style =
```

Comment:
Why this change is needed?

Reply:
Oh, I saw the linked issue. Won't CSS variable make it to have all Badges the same color?

Reply:
I do not understand this change. I think the issue here was, that we require `backgroundColor `property for it (so people cannot use className on top and override the value). Now we introduce some additional complexity with css variable?? Which as Vilius mentioned, gonna make all badges look same color.

Reply:
CSS variables are scoped, so every badge can have its own.

It does not fully address the problem, but it is the first step towards fixing the linked issue. Inline background-color is impossible to override using css without `!important` so this will help.

Reply:
I do not still understand, how this fixes the issue.. You still cannot use class.
Why not just make prop optional?

Reply:
> You still cannot use class.
?? Sure you can.

> Why not just make prop optional?
Because we don't have a default background-color, so badge looks broken.

Reply:
amm... idk, looks like inner hack to allow simple thing here..

Reply:
I don't see css variable as a hack.

Rather, I would say setting inline styles in a React-only package is more hacky.

Reply:
Well, are we planning to document anything about it? no.
By default we are setting css variable with an empty value, right?
So actually, we are providing from the CSS library the component, without background color.
Now, we are still leaving mandatory `backgroundColor `prop and then say what.. put class to override it? It just looks like something is wrong in there...

Reply:
Something is definitely wrong. That is why I said "this is the first step".

I was talking with @FlyersPh9 about setting a default value, and he suggested skyblue or ash, both of which seemed random to me. I think we could arrive at a more neutral color and use that as default.

Reply:
so whats the "second step" then? Just figure out the default color and make that optional? In that case, would we even need css variable?

Reply:
Yeah I think that is the second step, but @FlyersPh9 and I were also talking about redoing the whole data-viz color system 😅

CSS variable still helps in cases when you want to override the prop through CSS. Inline styles always make it hard.

Reply:
Converting to draft because I'm going to make some big changes in CSS first.

Reply:
Think this can be resolved now